# Problems 2.2

$$ \newcommand{\uvec}[1]{\hat{\mathbf{#1}}} $$

1. Construct geometrically: $\vec{OA}+\vec{OB}, \vec{OA}+\vec{BO}, -\vec{OA}+\vec{BO}$

<img src="Problems_2.2.1.svg">

2. State the necessary and sufficient conditions for the sum of two vectors to vanish.

    The two vectors must be of the same length (magnitude) and opposite direction. Since negation is geometrically equivalent to a 180 degree turn in direciton, opposite direction can arise either by switching endpoints or negating the vector.

3. Show by similar triangles, or otherwise, that $l(\vec a + \vec b) = l\vec a + l\vec b$.

    <img src="Problems_2.2.3.svg"></img>
    
    We can inspect the diagram above and see that the geometric interpretation of the distributive property of scalar multiplication makes intuitive sense. The angle between $\vec a$ and $\vec a+\vec b $ must be the same as the angle between $2\vec a$ and $2(\vec a + \vec b)$. Likewise, the angle between $\vec a$ and $\vec b$ must be the same as the angle between $2\vec a$ and $2\vec b$. With two angles proven to be the same, the third angle of a triangle must be the same since the interior angles sum to 180 degrees. It follows that the triangles formed by the bisection of the parallelograms in the diagram above must be similar, meaning they have the same angles, but only vary in size. Although the diagram establishes this by example with $l=2$, it's clear than $l$ should be able to have any real value because scalar multiplication of a vector can take any real value, and no real value can impact the angle of the vector (except $l=0$, which produces an undefined angle, but is a degenerate case for a triangle).
    
    ---

    If we decompose $a$ and $b$ into the ordered pairs $(x_a, y_a)$ and $(x_b, y_b)$ respectively, then their sum is the pair $(x_a+x_b, y_a+y_b)$. Scalar multiplication of $\vec a$ by $l$ yields the ordered pair $(l x_a, l y_a)$. Thus
    
    $$ l\vec a + l\vec b = (lx_a, ly_a) + (lx_b, ly_b) = (lx_a+lx_b, ly_a+ly_b) $$
    
    Since we already established that $\vec a + \vec b$ is equivalent to $(x_a+x_b, y_a+y+b)$, it follows by the distributive property that
    
    $$ l(\vec a + \vec b) = l(x_a+x_b, y_a+y_b) = (lx_a+lx_b, ly_a+ly_b) $$
    
    Which shows that $l(\vec a + \vec b)$ is equivalent to the same value as $l\vec a + l\vec b$

4. Show that the resolution of a vector along any three directions (where no two are parallel or antiparallel and all three do not lie in a plane) is unique. (Consider the consequences if this were not so).

    The vector composed of the ordered triplet $(x, y, z)$ is trivial to resolve to $x\uvec i + y\uvec j + z\uvec k$. If this vector somehow had multiple resolutions, that would imply that it refered to a point that had two positions along an axis, i.e. that the axes are looped. If we can take it as _given_ that any three basis vectors that are not parallel or coplanar are a linear transform of an orthonormal basis, then the same reasoning applies to the resolution along the transformed directions, just the same as the cartesian axes. (I don't feel like this is a satisfactory answer because that's a big _given_.)
    
    Maybe another way to think about it is in terms of infinitesimal wiggles. If we wiggle the coefficient of one of the basis vectors, then the vector moves even though the other two coefficients remain the same. There shouldn't be any way to wiggle the other coefficients to compensate for the displacement along the first 'axis', because they're independent. How to phrase that as a proof?

5. Find the moduli and direction cosines of the following vectors:

    $$ (-a, 0, 0), \quad (1, 1, -1), \quad (0, 1, -1), \quad (x, y, z) $$
    
    and of
    
    $$ [(-a, b, 0) + (a, b, 0)], \quad [(2, 3, -5) + (3, -2, 4)] $$
    
    and 
    
    $$ [(x, y, z) + (p, q, r)]. $$
    
    All symbols may be assumed to represent positive numbers.
    
    ---
    
    See p. 17-18 for the definition of the direction cosine. There's one for each axis, and it's the projection on that axis divided by the magnitude of the vector.
    
    $$ (x, y, z) = (rl, rm, rn) $$
    
    If we devide through by $r$ we get
    
    $$ (l, m, n) = \left(\frac{x}{r}, \frac{y}{r}, \frac{z}{r}\right) $$
    
    So moduli and direction cosines of the following:
    
    $$ \Vert(-a, 0, 0)\Vert = \sqrt{(-a)^2 + 0^2 +0^2} = a;\quad l = \frac{-a}{a} = -1,\ m = 0,\ n = 0 $$
    
    $$ \Vert(1, 1, -1)\Vert = \sqrt{1^2+1^2+(-1)^2} = \sqrt{3};\quad (l, m, n) = \left(\frac{1}{\sqrt{3}}, \frac{1}{\sqrt{3}}, \frac{-1}{\sqrt{3}}\right) $$
    
    $$ \Vert(0, 1, -1)\Vert = \sqrt{0^2 + 1^2 +(-1)^2} = \sqrt{2};\quad (l, m, n) = \left(0, \frac{1}{\sqrt 2}, \frac{-1}{\sqrt 2} \right)$$
    
    $$ \Vert(x, y, z)\Vert = \sqrt{x^2+y^2+z^2};\quad (l, m, n) = \left(\frac{x}{\sqrt{x^2+y^2+z^2}}, \frac{y}{\sqrt{x^2+y^2+z^2}}, \frac{z}{\sqrt{x^2+y^2+z^2}}\right) $$
    
    and of
    
    $$ [(-a, b, 0) + (a, b, 0)] = (-a+a, b+b, 0+0) = (0, 2b, 0) $$
    
    $$ \Vert(0, 2b, 0)\Vert = 2b; \quad (l, m, n) = (0, \frac{2b}{2\vert b\vert}, 0) = (0, 1, 0) $$
    
    $$ [(2, 3, -5) + (3, -2, 4)] = (5, 1, -1) $$
    
    $$ \Vert(5, 1, -1)\Vert = \sqrt{5^2, 1^2, (-1)^2} = \sqrt{25+1+1} = \sqrt{27} $$
    
    $$ (l, m, n) = \left(\frac{5}{\sqrt{27}}, \frac{1}{\sqrt{27}}, \frac{-1}{\sqrt{27}}\right) \approx (0.9623, 0.1925, -0.1925) $$
    
    and
    
    $$ [(x, y, z) + (p, q, r)] = (x+p, y+q, z+r) $$
    
    $$ \Vert(x+p, y+q, z+r)\Vert = \sqrt{(x+p)^2 + (y+q)^2 + (z+r)^2} $$
    
    $$ (l, m, n) = \left( \frac{x+p}{\sqrt{(x+p)^2 + (y+q)^2 + (z+r)^2}}, \frac{y+q}{\sqrt{(x+p)^2 + (y+q)^2 + (z+r)^2}}, \frac{z+r}{\sqrt{(x+p)^2 + (y+q)^2 + (z+r)^2}} \right) $$
    

6. Assuming that forces are vectors, find the resultant of two forces: $P_1$ along $(l_1, m_1, n_1)$ and $P_2$ along $(l_2, m_2, n_2)$.

    I feel like maybe I'm missing something here, but my expectation is that the sums of forces are just vector addition.

    $$ P_1 + P_2 = (l_1+l_2, m_1+m_2, n_1+n_2) $$
    
    If he means that $P_1$ and $P_2$ are scalars, maybe it could be
    
    $$ (P_1 l_1 + P_2 l_2, P_1 m_1 + P_2 m_2, P_1 n_1 + P_2 n_2) $$

7. Find the necessary and sufficent conditions for the sum of three nonzero vectors to vanish.

    In a geometric interpretation of summation, this means that the vectors laid tail to head in sequnce should come back around and point at the origin. Since there are three vectors, we know that they must form a triangle. The vectors will be coplanar, but that's not a necessary or sufficient condition, just a consequence of their triangle-ness. What is necessary is that the sum of any two is equal in magintude and opposite in direction to the third. Or, stated another way, the sum of their resolutions should be zero.

8. Show that

    (a) $\frac{(\vec{OA} + \vec{OB})}{2}$ is the mid-point of $\vec{AB}$;
    
    (b) $\frac{(n\vec{OA} + m\vec{OB})}{(m+n)}$ is the center of mass of masses $n$ at $A$ and $m$ at $B$;
    
    (c) $\frac{(\vec{OA} + \vec{OB} + \vec{OC})}{3}$ is the centroid of the triangle $ABC$.
    
    ---
    
    (a) The midpoint of $\vec{AB}$ is the average of the points $A$ and $B$. This is clear if we think of the midpoint as halfway between the vertical offset between $A$ and $B$ and, independently, halfway between the horizontal offset between $A$ and $B$. We can express this average as:
    
    $$ (\frac{x_a+x_b}{2}, \frac{y_a+y_b}{2}, \frac{z_a+z_b}{2}) $$
    
    If we expand the vector expression $\frac{(\vec{OA} + \vec{OB})}{2}$, we get
    
    $$ \frac{(x_a, y_a, z_a) + (x_b, y_b, z_b)}{2} =  \frac{(x_a+x_b, y_a+y_b, z_a+z_b)}{2} = (\frac{x_a+x_b}{2}, \frac{y_a+y_b}{2}, \frac{z_a+z_b}{2}) $$
    
    (b) With a simple scale having two masses and a balance point, if the ratio of the distances between the masses and the balance point is 2:1, then we can infer that the magnitude of the masses are also in a ratio of 2:1. The center of mass is an average of the positions of the two point masses, weighted by the respective masses. This is the physical interpretation of the equation given in the problem. The quantities $m$ and $n$ are the magnitudes of the masses. If $m = n$, then the center of mass should be equidistant between the masses, and we can see that in such cases, the formula is equivalent to a midpoint as established in (a)
    
    $$ \frac{(n\vec{OA} + m\vec{OB})}{(m+n)} = \frac{(m\vec{OA} + m\vec{OB})}{2m} = \frac{(mx_a, my_a, mz_a)+(mx_b, my_b, mz_b)}{2m} = \frac{m(x_a+x_b,y_a+y_b,z_a+z_b)}{2m} = \frac{(x_a+x_b,y_a+y_b,z_a+z_b)}{2} $$
    
    (c) The centroid of a triangle is defined as the arithmetic mean position of the points in a figure. So the proof here is asking whether one third the sum of these vectors is equivalent to an arithmetic mean. The arithmetic mean $A$ is in turn defined:
    
    $$ A = \frac{1}{n}\sum_{i=1}^{n} a_i $$
    
    Given $a_1=\vec{OA}$, $a_2=\vec{OB}$, and $a_3=\vec{OC}$, and $n=3$, it's clear that
    
    $$ A = \frac{1}{3}(\vec{OA}+\vec{OB}+\vec{OC}) = \frac{\vec{OA}+\vec{OB}+\vec{OC}}{3} $$

9. Prove by a vector method that the medians of a triangle are concurrent.

    Can we prove that the median of a triangle passes through the centroid? If all medians pass through the centroid, then the medians are concurrent because they all pass through one point. I think the general strategy should be to express the meidan as an equation and test if the centroid is a valid solution to that equation. Let's call the position of the centroid $\vec{OG}$, so
    
    $$ \vec{OG} = \frac{\vec{OA}+\vec{OB}+\vec{OC}}{3} $$

10. Two triangles $ABC$ and $A'B'C'$ have centroids $G$ and $G'$. Prove that

    $$ \vec{AA'} + \vec{BB'} + \vec{CC'} = 3\vec{GG'} $$
    
    <img src="Problems_2.2.10.svg">
    
    The vector $\vec{XX'} = \vec{OX'} - \vec{OX}$, where the point $O = (0, 0, 0)$. This allows us to express the centroids $OG$ and $OG'$ in terms of $\vec{OA}$, $\vec{OA'}$, $\vec{OB}$, etc.
    
    $$ \begin{align*}
        \vec{OG}  &= \frac{\vec{OA}+\vec{OB}+\vec{OC}}{3} \\
        \vec{OG'} &= \frac{\vec{OA'}+\vec{OB'}+\vec{OC'}}{3} \\
        \vec{GG'} &= \vec{OG'} - \vec{OG} \\
        \vec{GG'} &= \left(\frac{\vec{OA'}+\vec{OB'}+\vec{OC'}}{3}\right) - \left(\frac{\vec{OA}+\vec{OB}+\vec{OC}}{3}\right) \\
        3\vec{GG'} &= (\vec{OA'}-\vec{OA}) + (\vec{OB'} - \vec{OB}) + (\vec{OC'} - \vec{OC}) \\
        3\vec{GG'} &= \vec{AA'} + \vec{BB'} + \vec{CC'}
        \end{align*}
    $$

11. Express the vector equation of the line (2.2.1; below), in cartesians, and eliminate the parameter $\lambda$.

    $$ \vec r = \vec r_0 + \lambda\uvec i $$
    
    The answer here is a total theft ... [a cheat](https://tutorial.math.lamar.edu/Classes/CalcIII/EqnsOfLines.aspx).

    Note that $\vec r_0$ is a constant, and it is the exapanded values of $\uvec i$, also a constant, that uniquely shapes the slope of the plane. Let's make $\vec r = (x, y, z)$ and $\vec r_0 = (x_0, y_0, z_0)$, and we can assign arbitrary components such that the direction vector $\uvec i = (a, b, c)$.
    
    $$ (x, y, z) = (x_0, y_0, z_0) + \lambda (a, b, c) $$
    
    Since the three vectors' components are all also equal, this gives us three parametric equations:
    
    $$ \begin{align}
    x &= x_0 + \lambda a \\
    y &= y_0 + \lambda b \\
    z &= z_0 + \lambda c
    \end{align} $$
    
    All three equations share the common variable lambda, so we can eliminate lambda by solving for lambda and setting them equal
    
    $$ \frac{x-x_0}{a} = \frac{y-y_0}{b} = \frac{z-z_0}{c} $$


12. Show that the equation of a plane can be written

    $$ \vec r = \vec r_0 + \lambda \uvec a + \mu \uvec b $$
    
    where $\vec r_0$ is a point in the plane, $\uvec a$ and $\uvec b$ are any two nonparallel directions in the plane, and $\lambda$ and $\mu$ are variable scalars. Show that this is equivalent to the usual cartesian equation of the plane.
    
    [This](https://tutorial.math.lamar.edu/Classes/CalcIII/EqnsOfPlanes.aspx) doesn't seem to be as much help as it was with the lines...
    
    $$ \begin{align}
    x &= x_0 + \lambda x_a + \mu x_b \\
    y &= y_0 + \lambda y_a + \mu y_b \\
    z &= z_0 + \lambda z_a + \mu z_b
    \end{align}
    $$

If we add all three equations together...

$$ x + y + z = x_0 + y_0 + z_0 + \lambda x_a + \mu x_a ... $$

We can just call the right hand side $d$ and we have $ax + by + cz = d$.

13. If $Ox$ points toward the vernal equinox, and $Oz$ toward the celestial north pole, show that the direction cosines of a point $(\alpha, \delta)$ are

    $$ (\cos \alpha \cos \delta, \quad \sin \alpha \cos \delta, \quad \sin \delta). $$
    
    Find the direction cosines of the following points: the north pole of the ecliptic; the winter solstice; $(12\rm{h}, +45^\circ)$; $(15\rm{h}, -30^\circ)$; $(4\rm{h} \, 35\rm{m} \, 53.2\rm{s}, -58^\circ16'31'')$. In each case verify that the sum of the squares of the direction cosines is one.

In [7]:
import math

def hms(h, m=0, s=0):
    return math.pi * (h/12 + m/720 + s/43200)

def dms(d, m=0, s=0):
    return math.pi * (d/180 + m/10800 + s/648000)

def dircos(alpha, delta):
    '''Find the direction cosines of alpha (right ascension) and delta (declination).'''
    return (
        math.cos(alpha)*math.cos(delta),
        math.sin(alpha)*math.cos(delta),
        math.sin(delta)
    )

def dist(x, y, z):
    return math.sqrt(x**2 + y**2 + z**2)

The north pole of the ecliptic

In [8]:
dircos(hms(6), dms(90) + dms(23, 27)), dist(*dircos(hms(6), dms(90) + dms(23, 27)))

((-2.436732587779675e-17, -0.3979486313076102, 0.9174076993574883), 1.0)

The winter solstice is dipped 23 degrees 27 minutes below the equator in the "south" relative to the vernal equinox (negative 90 degrees).

In [54]:
dircos(hms(-6), -dms(23, 27)), dist(*dircos(dms(-90), -dms(23, 27)))

((5.617502012656427e-17, -0.9174076993574883, -0.3979486313076103), 1.0)

$(12\rm{h}, +45^\circ)$

In [34]:
dircos(hms(12), dms(45)), dist(*dircos(hms(12), dms(45)))

((-0.7071067811865476, 8.659560562354934e-17, 0.7071067811865475), 1.0)

In [35]:
dircos(hms(15), dms(-30)), dist(*dircos(hms(15), dms(-30)))

((-0.6123724356957947, -0.6123724356957945, -0.49999999999999994), 1.0)

In [36]:
dircos(hms(4, 35, 53.2), dms(-58, 16, 31)), dist(*dircos(hms(4, 35, 53.2), dms(-58, 16, 31)))

((0.19161053491220953, 0.4984256626933804, -0.8454923191131434), 1.0)

14. What is the equation of the plane of the ecliptic in the reference system of problem 13?

    Let's use the given vector parametric equation for the plane,

    $$ \vec r = \vec r_0 + \lambda \uvec a + \mu \uvec b $$
    
    We only need to provide two unit vectors in the plane, $\uvec a$ and $\uvec b$. Whe can specify these as the direction cosines of the x-axis and the solstice a quarter turn around the sphere from the vernal equinox. The x-axis is easy because we know its direction cosines directly:
    
    $$ \uvec a = (1, 0, 0) $$
    
    The value of $\uvec b$ can be found using $\alpha = 90^\circ$ and $\beta = 23^\circ 27'$.

In [37]:
dircos(dms(90), dms(23, 27))

(5.617502012656427e-17, 0.9174076993574883, 0.3979486313076103)

The first term rounds to zero, and we have

$$\uvec b = (0, 0.9174, 0.3979)$$
    
Plugging the vector values into the equation, we have
   
$$ \vec r = \vec r_0 + \lambda(1, 0, 0) + \mu(0, 0.9174, 0.3979) $$

15. If $Ox$ points toward the vernal equinox, and $Oz$ toward the north pole of the ecliptic, find the direction cosines of the celestial north and south poles, the autumnal equinox, and $(6\rm{h}, 0^\circ)$.

    We can think of the north pole as being at the point $(90^\circ, +90^\circ-23^\circ 27')$ and the south pole as being at the point $(90^\circ, -90^\circ-23^\circ 27')$

In [38]:
dircos(dms(90), dms(90)-dms(23, 27))

(2.436732587779676e-17, 0.3979486313076103, 0.9174076993574883)

In [39]:
dircos(dms(90), dms(-90)-dms(23, 27))

(-2.436732587779675e-17, -0.3979486313076102, -0.9174076993574883)

Similarly, we can think of the autumnal equinox as being opposite the vernal equinox, but we don't need to resort to trogonometry. Instead, we simply know that its $x$ component is the negation of the $x$ component of the vernal equinox, thus $(-1, 0, 0)$.

The last point, interpreted as a direction cosine in the reference frame of the ecliptic, need not resort to trigonometry either, it's simply one unit in the $y$ direction, i.e. $(0, 1, 0)$. However, if Danby is asking for the point $(6\rm h, 0^\circ)$ in the celestial system translated into the ecliptic coordinates, then we need to subtract the offset, much as we did for computing the location of the poles.

In [40]:
dircos(hms(6), -dms(23, 27))

(5.617502012656427e-17, 0.9174076993574883, -0.3979486313076103)

In [1]:
import astropy

In [16]:
from astropy.coordinates import spherical_to_cartesian
from astropy import units as u

In [49]:
x = spherical_to_cartesian(1*u.m, (23*u.deg + 27*u.arcmin), -90*u.deg)

In [52]:
type(x[0].value)

numpy.float64